UnityProjects/MRTKDevTemplate/Assets/Scripts/EyeTracking/FollowEyeGaze.cs

```diff
@@ -33,62 +32,56 @@ public class FollowEyeGaze : MonoBehaviour
-        private InputActionProperty _gazeTranslationAction;
+        [Tooltip("The IGazeInteractor that represents the gaze interaction.")]
+        private XRBaseInputInteractor gazeInteractor;
 
-        private IGazeInteractor gazeInteractor;
         private List<IXRInteractable> targets;
 
         private void Awake()
         {
             material = GetComponent<Renderer>().material;
-
-            gazeInteractor = gazeController.GetComponentInChildren<IGazeInteractor>();
-
             targets = new List<IXRInteractable>();
         }
 
-        private void OnEnable()
-        {
-            if (_gazeTranslationAction == null || _gazeTranslationAction.action == null)
-            {
-                return;
-            }
-
-            _gazeTranslationAction.action.performed += FollowEyeGazeAction;
-            _gazeTranslationAction.EnableDirectAction();
-        }
-        
-        private void OnDisable()
+        private void Update()
         {
-            if (_gazeTranslationAction == null || _gazeTranslationAction.action == null)
+            if (gazeInteractor == null)
             {
-                return;
+                return;                
             }
 
-            _gazeTranslationAction.DisableDirectAction();
-            _gazeTranslationAction.action.performed -= FollowEyeGazeAction;
-        }
-
-        private void Update()
-        {
             targets.Clear();
-
             gazeInteractor.GetValidTargets(targets);
             material.color = targets.Count > 0 ? hightlightStateColor : idleStateColor;
 
-            // Note: A better workflow would be to create and attach a prefab to the MRTK Gaze Controller object.
-            // Doing this will parent the cursor to the gaze controller transform and be updated automatically.
-            var pose = gazeController.transform.GetWorldPose();
-            transform.position = pose.position + gazeController.transform.forward * defaultDistanceInMeters;
+            if (TryGetGazeTransform(out Transform gazeTransform))
+            {
+                // Note: A better workflow would be to create and attach a prefab to the MRTK Gaze Controller object.
+                // Doing this will parent the cursor to the gaze controller transform and be updated automatically.
+                var pose = gazeTransform.GetWorldPose();
+                transform.position = pose.position + gazeTransform.forward * defaultDistanceInMeters;
+            }
         }
 
-        private void FollowEyeGazeAction(InputAction.CallbackContext obj)
+        /// <summary>
+        /// Attempt to obtain the gaze transform.
+        /// </summary>
+        private bool TryGetGazeTransform(out Transform transform)
         {
-            // Example of obtaining gaze input action properties
-            Vector3 translation = _gazeTranslationAction.action.ReadValue<Vector3>();
+            if (gazePoseDriver != null)
+            {
+                transform = gazePoseDriver.transform;
```

Comment:
```suggestion
                transform = gazeTrackedPoseDriver.transform;
```

Reply:
keeping as is...the "Tracked" seems too wordy